packages/components/autocomplete/stories/Autocomplete.stories.tsx

```diff
@@ -143,14 +148,22 @@ export const UsingGroupedItems = (args: AutocompleteProps<GroceryList>) => {
   const [filteredItems, setFilteredItems] = useState(groceryList);
 
   const handleInputValueChange = (value: string) => {
-    const newFilteredItems = groceryList.map((group) => {
-      return {
-        ...group,
-        options: group.options.filter((item: Produce) =>
+    const newFilteredItems = groceryList
```

Comment:
@gui-santos @Chaoste here is the needed change for the filter function. In this way the whole group entry gets removed and if there are no matches at all items is going to be completely empty and work just like any other "no items left in the list" case, no matter the shape of the items.

Reply:
Here would be another way of solving the filer which is maybe more elegant?
```
  const handleInputValueChange = (value: string) => {
    const newFilteredItems: GroceryList[] = [];
    groceryList.forEach((group) => {
      const filteredOptions = group.options.filter((item: Produce) =>
        item.name.toLowerCase().includes(value.toLowerCase()),
      );
      if (filteredOptions.length > 0) {
        newFilteredItems.push({
          groupTitle: group.groupTitle,
          options: filteredOptions,
        });
      }
    });
    setFilteredItems(newFilteredItems);
  };
```

Reply:
I didn't understand the problem

Why can't something like this work anymore?
```tsx
{!isLoading && items.length === 0 && (
  <div className={cx(styles.item, styles.disabled)}>
     {noMatchesMessage}
  </div>
)}
```

Reply:
I believe I fixed the reason. If the client only passes empty lists, we show the no matches string now.